server/src/main/java/org/candlepin/pinsetter/core/PinsetterTriggerListener.java

```diff
@@ -64,5 +70,23 @@ public void triggerMisfired(Trigger trigger) {
             trigger.getNextFireTime(),
             trigger.getMisfireInstruction(),
             trigger.getPriority());
+
+        String id = trigger.getJobKey().getName();
+        JobStatus js = jobCurator.find(id);
+        if (js == null) {
+            throw new RuntimeException("No JobStatus for job with id of " + id);
+        }
+        // if may not refire again, change to fail
+        if (!trigger.mayFireAgain()) {
+            log.warn("Job {} not allowed to fire again. Setting state to FAILED.", id);
+            js.setState(JobStatus.JobState.FAILED);
+            js.setResult("Failed run. Will not attempt again.");
+        }
+        else {
+            SimpleDateFormat sdf = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ssZ");
+            js.setResult("Will reattempt job at or after " + sdf.format(trigger.getNextFireTime()));
+            js.setState(JobStatus.JobState.PENDING);
+        }
+        jobCurator.merge(js);
```

Comment:
Quartz best practices state that all listener methods contain a try-catch and should handle ALL exceptions [1]. If something happens when committing the JobStatus, we violate this best practice and could potentially introduce issues.

[1] http://www.quartz-scheduler.org/documentation/best-practices.html#listeners-(triggerlistener-joblistener-schedulerlistener